edb/edgeql/compiler/expr.py

```diff
@@ -70,12 +71,18 @@ def compile_Path(
     return stmt.maybe_add_view(setgen.compile_path(expr, ctx=ctx), ctx=ctx)
 
 
+@dispatch.compile.register(qlast.SetUnionOp)
 @dispatch.compile.register(qlast.BinOp)
 def compile_BinOp(
         expr: qlast.BinOp, *, ctx: context.ContextLevel) -> irast.Set:
 
     op_node = func.compile_operator(
-        expr, op_name=expr.op, qlargs=[expr.left, expr.right], ctx=ctx)
+        expr,
+        op_name=expr.op,
+        qlargs=[expr.left, expr.right],
+        ctx=ctx,
+        setunion=isinstance(expr, qlast.SetUnionOp)
```

Comment:
Why not put this `isinstance` inside `compile_operator` instead of adding an argument?

Reply:
Done :man_facepalming: